### DISCUSSION TITLE
Reactive properties in TypeScript

### DISCUSSION THREAD

**Discussion by zuramai** [n6ai/minze#25] [Q&A] [answered]
Hello, I'm trying to change the value of a reactive properties in TypeScript, but I got error message like this:
![image](https://user-images.githubusercontent.com/45036724/156909010-e64bd933-d2fa-4caf-97cc-c537f4ec19d5.png)

```
Object is of type 'unknown'
```

This is my code:
```ts
import Minze, { EventListeners, MinzeElement, Reactive } from "minze"

export class CounterClick extends MinzeElement {
    reactive: Reactive = [
        ['count',0]
    ]


    onButtonClick = () => {
        this.count++
    }
  
    html = () => `
      <div class="counter">
        <div class="count">${this.count}</div>
        <m-button/>
      </div>
    `

    eventListeners: EventListeners = [
        [window, "minze:click", this.onButtonClick]
    ]

}
```

Any ideas how to fix this?

**Comment by sergejcodes** [comment, marked as answer]
All reactive properties are created dynamically, that's why TypeScript doesn't know their types. You have to explicitly define the types in a separate interface (that is exported with the same name as the component class). The docs have a more detailed [example](https://minze.dev/guide/advanced-typescript.html).

In your case it should look like this:
```ts
import Minze, { EventListeners, MinzeElement, Reactive } from "minze"

export interface CounterClick {
  count: number
}

export class CounterClick extends MinzeElement {
  reactive: Reactive = [
    ['count',0]
  ]
  
  // ...
}
```

**Reply by zuramai** [reply]
Ahh I see, I missed that part. Thank you for your reply!
